Significant Refactor

haqa opened the pull request:
Significant changes
 - Split monolithic module into separate files for each major class
 - Class/Method indentation tidying 
 - Manually merge MR #11 and MR #13 (correcting for new file locations)
 - Add command-line tools to dump data, set KVM parameters and PXE boot

Minor changes
 - Ignore changes to version.py if it happens to get imported (rewritable_git_version changes the file)
 - Add eclipse/pydev project files
 - Changed to my contact details

adrian-hungate-ocado committed:
Major refactor to enable sorting out dead code and unimplemented methods

adrian-hungate-ocado committed:
Add symlink to (in theory) allow tests to run

adrian-hungate-ocado committed:
Add simple command-line tools

adrian-hungate-ocado committed:
Manually merge change from MR #11

adrian-hungate-ocado committed:
Add KVM settings to wrykvmconfig

adrian-hungate-ocado committed:
Manually merge changes from MR #13

adrian-hungate-ocado committed:
Add missing decorator

adrian-hungate-ocado committed:
Ignore changes to verison.py (Avoid rewritable_git_version lossage)

adrian-hungate-ocado committed:
Ignore changes to verison.py (Avoid rewritable_git_version lossage)

Commit:
Correct help message

Commit:
Clean up unused imports and correct doccomment references

Commit:
Enumerate before get in dump() if available

adrian-hungate-ocado committed:
Remove unused code

adrian-hungate-ocado committed:
Remove debugging code

adrian-hungate-ocado committed:
Initial version of pure python wsman client

adrian-hungate-ocado committed:
Ignore test file

adrian-hungate-ocado committed:
Part way through switching to pure python wsman library

Eliminating magic, string bashing XML and guessing.
Most of the library is broken at this point.

adrian-hungate-ocado committed:
Migrated existing API to new underlying class

adrian-hungate-ocado committed:
Removed

adrian-hungate-ocado committed:
Separate data from classes and one class per module

adrian-hungate-ocado committed:
Added some unittests (Currently failing)

adrian-hungate-ocado committed:
Correct ignore and notice

adrian-hungate-ocado committed:
Fixed class re-use problem

adrian-hungate-ocado committed:
Factor out CLI common code to make CLI code simpler

adrian-hungate-ocado committed:
Clean up

adrian-hungate-ocado committed:
Add version. Note this will cause the file to be modified in your
working copy. This change is gitignored

adrian-hungate-ocado committed:
Add debug and showxml options for debugging

adrian-hungate-ocado committed:
Remove standalone code from test module

adrian-hungate-ocado committed:
Fix tests.

adrian-hungate-ocado committed:
Remove redundant import

adrian-hungate-ocado committed:
CLI tool to read and change power state

adrian-hungate-ocado committed:
Changed project settings for eclipse

adrian-hungate-ocado committed:
Added readme for tests

adrian-hungate-ocado committed:
Merge branch 'refactor' of https://github.com/ocadotechnology/wry into refactor

adrian-hungate-ocado committed:
Correct typo

adrian-hungate-ocado committed:
Correct typo

haqa committed:
Update README.md

haqa committed:
Update README.md

adrian-hungate-ocado committed:
More unit tests

adrian-hungate-ocado committed:
Make setting kvm params optional for wrykvmconfig. Display open ports.

adrian-hungate-ocado committed:
Update docs

adrian-hungate-ocado committed:
Add travis CI

adrian-hungate-ocado committed:
Add travis cache

adrian-hungate-ocado committed:
Add gitversion to requirements

adrian-hungate-ocado committed:
Fix bug where the RESOURCES dict was not instance local and the wrong machines got connected to

adrian-hungate-ocado committed:
Ignore build files

adrian-hungate-ocado committed:
Extend retries

adrian-hungate-ocado committed:
Return values from power state change commands

adrian-hungate-ocado committed:
Correct requirements

adrian-hungate-ocado committed:
Correct requirements for travis

adrian-hungate-ocado committed:
Remove use of gitversion as it breaks in many cases

haqa commented:
Last commits were to fix the tests. It transpires that "gitversion" is a little fragile so I've removed it (for now?)

haqa committed:
Add build status flag from TravisCI

adrian-hungate-ocado committed:
Add support for reading ethernet hardware configuration

mikebryant commented on tests/TestBase.py:
```diff
@@ -0,0 +1,79 @@
+'''
+Created on 10 Jul 2017
+
+@author: adrian
```
We don't normally include headers like these - that's what git is for

mikebryant replied:
(This comment applies to all the files)

haqa replied:
I'll remove them and update my Eclipse template for this project.

mikebryant commented on tests/TestBase.py:
```diff
@@ -0,0 +1,79 @@
+#        print len(data)
+        data = self.UUID.sub("uuid:", data)
+#        print len(data)
+#        print len(req)
```
We shouldn't be committing commented code like this

haqa replied:
Changed to conditional on a flag (With explanatory comment)

mikebryant commented on wry/AMTDevice.py:
```diff
@@ -0,0 +1,118 @@
+        messages = ['# Could not dump %s' % name for name in impossible]
+        if as_json:
+            return '\n'.join(messages) + '\n' + output.as_json()
+        print '\n'.join(messages)
```
We probably shouldn't have prints in library code

haqa replied:
This a dump method who's doc-comment states it's purpose - To print that current state of AMT settings

haqa replied:
Removed

mikebryant commented on wry/AMTKVM.py:
```diff
@@ -0,0 +1,248 @@
+}
+
+
+#class EnablementMap(object):
```
Don't commit this, if it's not ready or used?

haqa replied:
It should have been removed. It has now.

mikebryant commented on wry/AMTKVM.py:
```diff
@@ -0,0 +1,248 @@
+                ports.add(16995)
+        return ports
+#
+#    @enabled_ports.setter
```
Is this chunk supposed to be here? Or can it be deleted?

haqa replied:
I've deleted it

mikebryant commented on wry/wsmanResource.py:
```diff
@@ -0,0 +1,213 @@
+        else:
+            doc = doc % params
+        if self.showxml:
+            print "==== Request ===="
```
prints?

haqa replied:
Debugging code to allow users of the library to dump the conversations. Enabled only if self.showxml is set

adrian-hungate-ocado committed:
Changes requested by Mike as part of code review

Remove file date/author comments
Change commented prints in tests to optional blocks
Remove multiple blocks of unused code

adrian-hungate-ocado committed:
Bump library version to 1.0

adrian-hungate-ocado committed:
Remove debug code

adrian-hungate-ocado committed:
Update doc comment

mikebryant commented:
cc @godley could you help me take a look at this?
